### PR TITLE
Fix `or` deadlock issue.

### PR DESCRIPTION
**Pull request by cheatfate**
Address issue #516.

### CODE REVIEW

File: chronos/internal/asyncfutures.nim [hunk 1]
```diff
@@ -1667,8 +1667,7 @@ proc `or`*[T, Y, E1, E2](
   type
     InternalRaisesFutureRaises = union(E1, E2)
 
-  let
-    retFuture = newFuture[void]("chronos.wait()", {FutureFlag.OwnCancelSchedule})
+  let retFuture = newFuture[void]("chronos.or()", {})
```
Comment (arnetheduck) [inline chronos/internal/asyncfutures.nim:1670]: `cancelSoon(fut or fut2)` will defect in this case?

Comment (cheatfate) [reply chronos/internal/asyncfutures.nim:1670]: why?

Comment (arnetheduck) [reply chronos/internal/asyncfutures.nim:1670]: nevermind, my mistake

File: chronos/internal/asyncfutures.nim [hunk 2]
```diff
@@ -1667,8 +1667,7 @@ proc `or`*[T, Y, E1, E2](
   type
     InternalRaisesFutureRaises = union(E1, E2)
```
Comment (arnetheduck) [inline chronos/internal/asyncfutures.nim:1668]: this might be a problem if neither of E1, E2 raise `CancelledError`

Comment (cheatfate) [reply chronos/internal/asyncfutures.nim:1668]: How i can do this? Because looks like `prepend` and `union` are not working here.

Comment (cheatfate) [reply chronos/internal/asyncfutures.nim]: Fixed in https://github.com/status-im/nim-chronos/pull/517/commits/d5fd6663b2997328fda2a63aff9c9f69955c5b3f

**Pull request merged**